In [1]:
import pandas as pd
import numpy as np
from scipy import stats

# Data Transaksi

In [2]:
data = pd.read_excel('https://github.com/fahmimnalfrzki/Additional-Materials/blob/main/online_retail_data.xlsx?raw=true').drop(columns='Unnamed: 0').sort_values('InvoiceDate',ascending=True).reset_index(drop=True)
data

InvoiceNo StockCode                        Description  Quantity  \
0       536369     21756           BATH BUILDING BLOCK WORD         3   
1       536373     82482  WOODEN PICTURE FRAME WHITE FINISH         6   
2       536398     22752       SET 7 BABUSHKA NESTING BOXES         6   
3       536401     85150      LADIES & GENTLEMEN METAL SIGN         1   
4       536404     20727            LUNCH BAG  BLACK SKULL.        10   
...        ...       ...                                ...       ...   
9995    581501     22942        CHRISTMAS LIGHTS 10 SANTAS         12   
9996    581572     22627                MINT KITCHEN SCALES         4   
9997    581578     21507            ELEPHANT BIRTHDAY CARD         12   
9998    581579     23353     6 GIFT TAGS VINTAGE CHRISTMAS         12   
9999    581580    85049E          SCANDINAVIAN REDS RIBBONS         2   

             InvoiceDate  UnitPrice  CustomerID         Country  
0    2010-12-01 08:35:00       5.95     13047.0  United Kingdom  
1    2010-12-01 09:02:00       2.10     17850.0  United Kingdom  
2    2010-12-01 10:52:00       8.50     13448.0  United Kingdom  
3    2010-12-01 11:21:00       2.55     15862.0  United Kingdom  
4    2010-12-01 11:29:00       1.65     16218.0  United Kingdom  
...                  ...        ...         ...             ...  
9995 2011-12-09 10:46:00       2.95     12985.0  United Kingdom  
9996 2011-12-09 12:08:00       8.50     16705.0  United Kingdom  
9997 2011-12-09 12:16:00       0.42     12713.0         Germany  
9998 2011-12-09 12:19:00       0.83     17581.0  United Kingdom  
9999 2011-12-09 12:20:00       1.25     12748.0  United Kingdom  

[10000 rows x 8 columns]

## 1. Menentukan Stok Barang di tahun berikutnya

Perusahaan Anda berencana membuat gudang baru untuk meningkatkan pelayanan pengiriman barang yang kian membludak. Apakah Anda bisa menentukan berapa kapasitas gudang yang seharusnya dibuat? (asumsi bahwa kapasitas gudang ditentukan dari berapa banyak barang yang keluar tiap harinya)


In [3]:
# ekstraksi tanggal dari kolom InvoiceDate
data['tanggal'] = data['InvoiceDate'].dt.date
data

InvoiceNo StockCode                        Description  Quantity  \
0       536369     21756           BATH BUILDING BLOCK WORD         3   
1       536373     82482  WOODEN PICTURE FRAME WHITE FINISH         6   
2       536398     22752       SET 7 BABUSHKA NESTING BOXES         6   
3       536401     85150      LADIES & GENTLEMEN METAL SIGN         1   
4       536404     20727            LUNCH BAG  BLACK SKULL.        10   
...        ...       ...                                ...       ...   
9995    581501     22942        CHRISTMAS LIGHTS 10 SANTAS         12   
9996    581572     22627                MINT KITCHEN SCALES         4   
9997    581578     21507            ELEPHANT BIRTHDAY CARD         12   
9998    581579     23353     6 GIFT TAGS VINTAGE CHRISTMAS         12   
9999    581580    85049E          SCANDINAVIAN REDS RIBBONS         2   

             InvoiceDate  UnitPrice  CustomerID         Country     tanggal  
0    2010-12-01 08:35:00       5.95     13047.0  United Kingdom  2010-12-01  
1    2010-12-01 09:02:00       2.10     17850.0  United Kingdom  2010-12-01  
2    2010-12-01 10:52:00       8.50     13448.0  United Kingdom  2010-12-01  
3    2010-12-01 11:21:00       2.55     15862.0  United Kingdom  2010-12-01  
4    2010-12-01 11:29:00       1.65     16218.0  United Kingdom  2010-12-01  
...                  ...        ...         ...             ...         ...  
9995 2011-12-09 10:46:00       2.95     12985.0  United Kingdom  2011-12-09  
9996 2011-12-09 12:08:00       8.50     16705.0  United Kingdom  2011-12-09  
9997 2011-12-09 12:16:00       0.42     12713.0         Germany  2011-12-09  
9998 2011-12-09 12:19:00       0.83     17581.0  United Kingdom  2011-12-09  
9999 2011-12-09 12:20:00       1.25     12748.0  United Kingdom  2011-12-09  

[10000 rows x 9 columns]

In [4]:
data.describe()

Quantity     UnitPrice    CustomerID
count  10000.000000  10000.000000   7499.000000
mean       9.719000      4.975422  15275.863582
std       37.130479     63.828081   1708.921268
min    -1131.000000      0.000000  12347.000000
25%        1.000000      1.250000  13930.500000
50%        3.000000      2.080000  15144.000000
75%       12.000000      4.130000  16779.000000
max     1152.000000   5522.140000  18287.000000

In [5]:
# Cara mengambil data tanggal atau waktu (tinggal mengganti 'date' atau 'time')
data['InvoiceDate'].dt.date

0       2010-12-01
1       2010-12-01
2       2010-12-01
3       2010-12-01
4       2010-12-01
           ...    
9995    2011-12-09
9996    2011-12-09
9997    2011-12-09
9998    2011-12-09
9999    2011-12-09
Name: InvoiceDate, Length: 10000, dtype: object

In [6]:
# Ekstraksi tanggal dari kolom InvoiceDate
data['Tanggal']=data['InvoiceDate'].dt.date
data

InvoiceNo StockCode                        Description  Quantity  \
0       536369     21756           BATH BUILDING BLOCK WORD         3   
1       536373     82482  WOODEN PICTURE FRAME WHITE FINISH         6   
2       536398     22752       SET 7 BABUSHKA NESTING BOXES         6   
3       536401     85150      LADIES & GENTLEMEN METAL SIGN         1   
4       536404     20727            LUNCH BAG  BLACK SKULL.        10   
...        ...       ...                                ...       ...   
9995    581501     22942        CHRISTMAS LIGHTS 10 SANTAS         12   
9996    581572     22627                MINT KITCHEN SCALES         4   
9997    581578     21507            ELEPHANT BIRTHDAY CARD         12   
9998    581579     23353     6 GIFT TAGS VINTAGE CHRISTMAS         12   
9999    581580    85049E          SCANDINAVIAN REDS RIBBONS         2   

             InvoiceDate  UnitPrice  CustomerID         Country     tanggal  \
0    2010-12-01 08:35:00       5.95     13047.0  United Kingdom  2010-12-01   
1    2010-12-01 09:02:00       2.10     17850.0  United Kingdom  2010-12-01   
2    2010-12-01 10:52:00       8.50     13448.0  United Kingdom  2010-12-01   
3    2010-12-01 11:21:00       2.55     15862.0  United Kingdom  2010-12-01   
4    2010-12-01 11:29:00       1.65     16218.0  United Kingdom  2010-12-01   
...                  ...        ...         ...             ...         ...   
9995 2011-12-09 10:46:00       2.95     12985.0  United Kingdom  2011-12-09   
9996 2011-12-09 12:08:00       8.50     16705.0  United Kingdom  2011-12-09   
9997 2011-12-09 12:16:00       0.42     12713.0         Germany  2011-12-09   
9998 2011-12-09 12:19:00       0.83     17581.0  United Kingdom  2011-12-09   
9999 2011-12-09 12:20:00       1.25     12748.0  United Kingdom  2011-12-09   

         Tanggal  
0     2010-12-01  
1     2010-12-01  
2     2010-12-01  
3     2010-12-01  
4     2010-12-01  
...          ...  
9995  2011-12-09  
9996  2011-12-09  
9997  2011-12-09  
9998  2011-12-09  
9999  2011-12-09  

[10000 rows x 10 columns]

In [7]:
# Menghilangkan nilai quantity yang minus
data_update = data[data['Quantity'] > 0]
data_update

InvoiceNo StockCode                        Description  Quantity  \
0       536369     21756           BATH BUILDING BLOCK WORD         3   
1       536373     82482  WOODEN PICTURE FRAME WHITE FINISH         6   
2       536398     22752       SET 7 BABUSHKA NESTING BOXES         6   
3       536401     85150      LADIES & GENTLEMEN METAL SIGN         1   
4       536404     20727            LUNCH BAG  BLACK SKULL.        10   
...        ...       ...                                ...       ...   
9995    581501     22942        CHRISTMAS LIGHTS 10 SANTAS         12   
9996    581572     22627                MINT KITCHEN SCALES         4   
9997    581578     21507            ELEPHANT BIRTHDAY CARD         12   
9998    581579     23353     6 GIFT TAGS VINTAGE CHRISTMAS         12   
9999    581580    85049E          SCANDINAVIAN REDS RIBBONS         2   

             InvoiceDate  UnitPrice  CustomerID         Country     tanggal  \
0    2010-12-01 08:35:00       5.95     13047.0  United Kingdom  2010-12-01   
1    2010-12-01 09:02:00       2.10     17850.0  United Kingdom  2010-12-01   
2    2010-12-01 10:52:00       8.50     13448.0  United Kingdom  2010-12-01   
3    2010-12-01 11:21:00       2.55     15862.0  United Kingdom  2010-12-01   
4    2010-12-01 11:29:00       1.65     16218.0  United Kingdom  2010-12-01   
...                  ...        ...         ...             ...         ...   
9995 2011-12-09 10:46:00       2.95     12985.0  United Kingdom  2011-12-09   
9996 2011-12-09 12:08:00       8.50     16705.0  United Kingdom  2011-12-09   
9997 2011-12-09 12:16:00       0.42     12713.0         Germany  2011-12-09   
9998 2011-12-09 12:19:00       0.83     17581.0  United Kingdom  2011-12-09   
9999 2011-12-09 12:20:00       1.25     12748.0  United Kingdom  2011-12-09   

         Tanggal  
0     2010-12-01  
1     2010-12-01  
2     2010-12-01  
3     2010-12-01  
4     2010-12-01  
...          ...  
9995  2011-12-09  
9996  2011-12-09  
9997  2011-12-09  
9998  2011-12-09  
9999  2011-12-09  

[9812 rows x 10 columns]

In [8]:
# Group by berdasarkan tanggal dan quantity
stock = data_update[['Tanggal','Quantity']].groupby('Tanggal').sum()
stock

Quantity
Tanggal             
2010-12-01       364
2010-12-02       487
2010-12-03       244
2010-12-05       212
2010-12-06       292
...              ...
2011-12-05       695
2011-12-06       641
2011-12-07       519
2011-12-08       923
2011-12-09        98

[305 rows x 1 columns]

In [9]:
stock.describe()

Quantity
count   305.000000
mean    335.809836
std     252.767192
min      15.000000
25%     179.000000
50%     258.000000
75%     415.000000
max    1463.000000

In [10]:
print('Rata-Rata =', stock['Quantity'].mean())
print('Maksimum =', stock['Quantity'].max())
print('Minimum=', stock['Quantity'].min())

Rata-Rata = 335.8098360655738
Maksimum = 1463
Minimum= 15


Dalam setahun kebelakang, jumlah barang yang keluar dari gudang mulai dari 15-1463 barang. Rerata dalam sehari barang keluar dari gudang sebesar 336 barang.

#Menghitung confident interval dengan asumsi data populasi terdistribusi normal
#Rumus
#stats.norm.interval(conf_level, loc=average, scale="lebar_distribusi_dari_average")

#scale merupakan lebar distribusidari average =? std/sqrt(N)

In [11]:
std = stock['Quantity'].std()
N = len(stock)
stats.norm.interval(0.95, loc=stock['Quantity'].mean(), scale=std/np.sqrt(N))

(307.4424790380824, 364.17719309306517)

Diperkirakan jumlah barang yang memungkinkan keluar setiap harinya yakni 307 sampai 364 barang.

Rekomendasi: Buat gudang dengan kapasitas barang yang muat hingga 364 atau 400 barang.

## 2. Mana yang lebih baik? Penjualan di Jerman atau Inggris?

Anda diminta untuk menganalisa performa penjualan di Jerman dan Inggris. Penjualan yang dimaksud adalah pendapatan bulanan yang berasal dari kedua negara tersebut. Tujuan analisis ini ingin menentukan lokasi cabang baru.

In [12]:
data['Bulan'] = data['InvoiceDate'].dt.month
data['Tahun'] = data['InvoiceDate'].dt.year
data['Revenue'] = data['Quantity']*data['UnitPrice']
data

InvoiceNo StockCode                        Description  Quantity  \
0       536369     21756           BATH BUILDING BLOCK WORD         3   
1       536373     82482  WOODEN PICTURE FRAME WHITE FINISH         6   
2       536398     22752       SET 7 BABUSHKA NESTING BOXES         6   
3       536401     85150      LADIES & GENTLEMEN METAL SIGN         1   
4       536404     20727            LUNCH BAG  BLACK SKULL.        10   
...        ...       ...                                ...       ...   
9995    581501     22942        CHRISTMAS LIGHTS 10 SANTAS         12   
9996    581572     22627                MINT KITCHEN SCALES         4   
9997    581578     21507            ELEPHANT BIRTHDAY CARD         12   
9998    581579     23353     6 GIFT TAGS VINTAGE CHRISTMAS         12   
9999    581580    85049E          SCANDINAVIAN REDS RIBBONS         2   

             InvoiceDate  UnitPrice  CustomerID         Country     tanggal  \
0    2010-12-01 08:35:00       5.95     13047.0  United Kingdom  2010-12-01   
1    2010-12-01 09:02:00       2.10     17850.0  United Kingdom  2010-12-01   
2    2010-12-01 10:52:00       8.50     13448.0  United Kingdom  2010-12-01   
3    2010-12-01 11:21:00       2.55     15862.0  United Kingdom  2010-12-01   
4    2010-12-01 11:29:00       1.65     16218.0  United Kingdom  2010-12-01   
...                  ...        ...         ...             ...         ...   
9995 2011-12-09 10:46:00       2.95     12985.0  United Kingdom  2011-12-09   
9996 2011-12-09 12:08:00       8.50     16705.0  United Kingdom  2011-12-09   
9997 2011-12-09 12:16:00       0.42     12713.0         Germany  2011-12-09   
9998 2011-12-09 12:19:00       0.83     17581.0  United Kingdom  2011-12-09   
9999 2011-12-09 12:20:00       1.25     12748.0  United Kingdom  2011-12-09   

         Tanggal  Bulan  Tahun  Revenue  
0     2010-12-01     12   2010    17.85  
1     2010-12-01     12   2010    12.60  
2     2010-12-01     12   2010    51.00  
3     2010-12-01     12   2010     2.55  
4     2010-12-01     12   2010    16.50  
...          ...    ...    ...      ...  
9995  2011-12-09     12   2011    35.40  
9996  2011-12-09     12   2011    34.00  
9997  2011-12-09     12   2011     5.04  
9998  2011-12-09     12   2011     9.96  
9999  2011-12-09     12   2011     2.50  

[10000 rows x 13 columns]

In [13]:
data_update = data[data['Quantity'] > 0]
data_update

InvoiceNo StockCode                        Description  Quantity  \
0       536369     21756           BATH BUILDING BLOCK WORD         3   
1       536373     82482  WOODEN PICTURE FRAME WHITE FINISH         6   
2       536398     22752       SET 7 BABUSHKA NESTING BOXES         6   
3       536401     85150      LADIES & GENTLEMEN METAL SIGN         1   
4       536404     20727            LUNCH BAG  BLACK SKULL.        10   
...        ...       ...                                ...       ...   
9995    581501     22942        CHRISTMAS LIGHTS 10 SANTAS         12   
9996    581572     22627                MINT KITCHEN SCALES         4   
9997    581578     21507            ELEPHANT BIRTHDAY CARD         12   
9998    581579     23353     6 GIFT TAGS VINTAGE CHRISTMAS         12   
9999    581580    85049E          SCANDINAVIAN REDS RIBBONS         2   

             InvoiceDate  UnitPrice  CustomerID         Country     tanggal  \
0    2010-12-01 08:35:00       5.95     13047.0  United Kingdom  2010-12-01   
1    2010-12-01 09:02:00       2.10     17850.0  United Kingdom  2010-12-01   
2    2010-12-01 10:52:00       8.50     13448.0  United Kingdom  2010-12-01   
3    2010-12-01 11:21:00       2.55     15862.0  United Kingdom  2010-12-01   
4    2010-12-01 11:29:00       1.65     16218.0  United Kingdom  2010-12-01   
...                  ...        ...         ...             ...         ...   
9995 2011-12-09 10:46:00       2.95     12985.0  United Kingdom  2011-12-09   
9996 2011-12-09 12:08:00       8.50     16705.0  United Kingdom  2011-12-09   
9997 2011-12-09 12:16:00       0.42     12713.0         Germany  2011-12-09   
9998 2011-12-09 12:19:00       0.83     17581.0  United Kingdom  2011-12-09   
9999 2011-12-09 12:20:00       1.25     12748.0  United Kingdom  2011-12-09   

         Tanggal  Bulan  Tahun  Revenue  
0     2010-12-01     12   2010    17.85  
1     2010-12-01     12   2010    12.60  
2     2010-12-01     12   2010    51.00  
3     2010-12-01     12   2010     2.55  
4     2010-12-01     12   2010    16.50  
...          ...    ...    ...      ...  
9995  2011-12-09     12   2011    35.40  
9996  2011-12-09     12   2011    34.00  
9997  2011-12-09     12   2011     5.04  
9998  2011-12-09     12   2011     9.96  
9999  2011-12-09     12   2011     2.50  

[9812 rows x 13 columns]

In [14]:
UK = data_update[data_update['Country']=='United Kingdom'][['Tahun','Bulan','Revenue']].groupby(['Tahun','Bulan']).sum()
DE = data_update[data_update['Country']=='Germany'][['Tahun','Bulan','Revenue']].groupby(['Tahun','Bulan']).sum()
print(UK), print(DE)

              Revenue
Tahun Bulan          
2010  12     13367.49
2011  1      10109.07
      2       7018.60
      3      10230.26
      4       8442.48
      5      11904.22
      6      11666.44
      7       9546.05
      8      12053.97
      9      16268.74
      10     18028.77
      11     24926.92
      12      7782.09
             Revenue
Tahun Bulan         
2010  12      309.10
2011  1       326.51
      2        42.90
      3       546.20
      4       368.68
      5       460.72
      6       134.50
      7       164.19
      8       148.80
      9       231.59
      10      596.20
      11      437.70
      12       82.58


(None, None)

In [15]:
print('Rata-rata pendapatan bulanan UK =', UK['Revenue'].mean())
print('Rata-rata pendapatan bulanan Germany =', DE['Revenue'].mean())

Rata-rata pendapatan bulanan UK = 12411.16153846154
Rata-rata pendapatan bulanan Germany = 296.1284615384615


Penjualan rata-rata di UK jauh lebih tinggi dibandingkan Germany. Artinya performa penjualan di UK lebih baik pada satu tahun ke belakang.

Melakukan Uji Hipotesis Two Sample Independent
==> karena ingin menguji perbedaan signifikan "data populasi" antara penjualan di UK dan Germany dari data sampel setahun kebelakang

H0: Rata-rata pendapatan bulanan UK <= Germany

H1: Rata-rata pendapatan bulanan UK > Germany 

In [16]:
t_stats, p_val=stats.ttest_ind(UK['Revenue'],DE['Revenue'])
print('T-Statistic:', t_stats)
print('P-value:', p_val/2) # ==> dibagi 2 karena one sided

T-Statistic: 8.88800098677858
P-value: 2.3313412556798145e-09


critical value = 0.05

p-value = 2.33 x 10^-9

critical value < p-value


\#Kesimpulan
Dari pengujian statistik didapatkan bahwasanya penjulan di UK akan lebih well-performed dibandingkan di Germany. Maka yang paling bagus untuk membuka cabang di UK untuk menggait pembeli yang berasal dari Geramany meilhat posisi geografis juga lebih dekat.

# Data Asuransi

In [17]:
ins = pd.read_csv('https://github.com/fahmimnalfrzki/Dataset/raw/main/insurance.csv')
ins

age     sex     bmi  children smoker     region      charges
0      19  female  27.900         0    yes  southwest  16884.92400
1      18    male  33.770         1     no  southeast   1725.55230
2      28    male  33.000         3     no  southeast   4449.46200
3      33    male  22.705         0     no  northwest  21984.47061
4      32    male  28.880         0     no  northwest   3866.85520
...   ...     ...     ...       ...    ...        ...          ...
1333   50    male  30.970         3     no  northwest  10600.54830
1334   18  female  31.920         0     no  northeast   2205.98080
1335   18  female  36.850         0     no  southeast   1629.83350
1336   21  female  25.800         0     no  southwest   2007.94500
1337   61  female  29.070         0    yes  northwest  29141.36030

[1338 rows x 7 columns]

## 1. Apakah Ada Hubungan Antara Gender dengan Kebiasaan Merokok?

Melakukan uji hipotesis dengan tipe chi-square

H0: Tidak ada hubungan antara Gender dengan Kebiasaan Merokok (Pria dan Wanita sama)

H1 : Ada hubungan antara Gender dengan Kebiasaan Merokok (Pria dan Wanita berbeda)

In [18]:
# Membuat tabel kontigensi / contingency table
contingency_table = pd.crosstab(ins['sex'], ins['smoker'])
contingency_table

smoker   no  yes
sex             
female  547  115
male    517  159

In [19]:
stat, p_val, dof, expected = stats.chi2_contingency(contingency_table)
print("P-value:",p_val)

P-value: 0.006548143503580696


Critical value: 0.05

p-value = 0.005 < critical value

H0 ditolak

**Kesimpulan**: 


Ada hubungan antara jenis kelamin dengan kebiasaan merokok, dimana pria yang merokok lebih besar dibandingkan wanita.

## 2. Apakah Biaya Premi Laki-laki dan Perempuan Sama?

In [20]:
pria = ins[ins['sex']=='male']['charges']
wanita = ins[ins['sex']=='female']['charges']

print("Rata-rata premi pria:", pria.mean())
print("Rata-rata premi wanita:", wanita.mean())

Rata-rata premi pria: 13956.751177721893
Rata-rata premi wanita: 12569.578843835347


Melakukan Uji Hipotesis Two Sample Independent Two Sided

H0: Rata-rata premi pria = premi wanita

H1: Rata-rata premi pria != premi wanita

In [21]:
t_stat, p_val = stats.ttest_ind(pria,wanita)
print("P-value:", p_val)

P-value: 0.03613272100592976


# Tugas (PR)


Kamu adalah anggota tim Data Science di Austin Smart City dan sedang mengerjakan projek analisa sistem pembuangan dan pengelolaan sampah di kota tersebut. Sebagai anggota tim, berdasarkan data yang tersedia, coba analisa data tersebut menggunakan statistik deskriptif dan inferensial, serta berikan insight kepada pemerintah kota Austin mengenai kondisi sistem pembuangan dan pengelolaan sampah. Gunakan guideline/langkah berikut untuk mempermudah kamu dalam menganalisa.

1. Pemerintah kota berencana menambah site baru. Berikan analisa dan saran terkait berapa kapasitas penampungan di site baru berdasarkan perhitungan `confidence interval`. Data apa yang kamu gunakan, apakah hanya dari satu site saja atau dari keseluruhan site?

2. Buatlah `analisa uji hipotesis` dari data tersebut dari data yang tersedia (kasusnya dibebaskan). Definisikan H0 dan H1 nya serta lakukan perhitungan menggunakan t-test yang sesuai dengan hipotesis kamu. Tulis kesimpulan dari hasil perhitunganmu kepada pemerintah kota (**Ingat!** pemerintah kota tidak mengerti p-value, hipotesis diterima/tidak diterima, jadi gunakan bahasa yang lebih manusiawi!).

Akses Data: https://github.com/fahmimnalfrzki/Dataset/raw/main/data_tugas.csv

In [22]:
ds = pd.read_csv('https://github.com/fahmimnalfrzki/Dataset/raw/main/data_tugas.csv')
ds

report_date                       load_time load_type  load_weight  \
0      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
1      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
2      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
3      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
4      2021-01-02  2021-01-02 09:24:00.000000 UTC     TIRES       1640.0   
...           ...                             ...       ...          ...   
26152  2021-07-09  2021-07-09 09:48:00.000000 UTC     BRUSH       8240.0   
26153  2021-07-09  2021-07-09 12:02:00.000000 UTC     BRUSH       4160.0   
26154  2021-07-09  2021-07-09 11:55:00.000000 UTC     BRUSH       4340.0   
26155  2021-07-09  2021-07-09 12:00:00.000000 UTC    LITTER       1900.0   
26156  2021-07-09  2021-07-09 12:00:00.000000 UTC    LITTER       3040.0   

       dropoff_site  
0       ONION CREEK  
1       ONION CREEK  
2       ONION CREEK  
3       ONION CREEK  
4               MRF  
...             ...  
26152  HORNSBY BEND  
26153  HORNSBY BEND  
26154  HORNSBY BEND  
26155  TDS LANDFILL  
26156  TDS LANDFILL  

[26157 rows x 5 columns]

In [23]:
print(ds['load_type'].unique())
print(ds['dropoff_site'].unique())

['SWEEPING' 'TIRES' 'RECYCLING - SINGLE STREAM' 'YARD TRIMMING' 'BULK'
 'DEAD ANIMAL' 'GARBAGE COLLECTIONS' 'MIXED LITTER' 'ORGANICS' 'BRUSH'
 'RECYCLED METAL' 'LITTER' 'CONTAMINATED RECYCLING'
 'CONTAMINATED ORGANICS']
['ONION CREEK' 'MRF' 'TDS - MRF' 'HORNSBY BEND' 'TDS LANDFILL'
 'ORGANICS BY GOSH' 'BALCONES RECYCLING' 'AUSTIN IRON AND METAL'
 'PARK CENTER' 'ON ROUTE HIGH DUMP' 'WILCAB' 'GREAT NORTHERN'
 'AUSTIN WOOD RECYCLING' 'PARK BEND' 'BALCONES WOODS' 'STEINER LANDFILL'
 'WESTFIELD' 'ST. EDWARDS']


In [24]:
ds.groupby(['load_type','dropoff_site'])['load_weight'].agg(['min','mean','max'])

min          mean  \
load_type                 dropoff_site                                      
BRUSH                     HORNSBY BEND              60.0000   6230.978836   
                          TDS LANDFILL            6580.0000   9220.000000   
BULK                      MRF                    12420.0000  13750.000000   
                          TDS - MRF               4200.0000   9880.000000   
                          TDS LANDFILL              20.0000   8109.106918   
CONTAMINATED ORGANICS     TDS LANDFILL           25380.0000  25380.000000   
CONTAMINATED RECYCLING    TDS LANDFILL            5160.0000   5160.000000   
DEAD ANIMAL               ORGANICS BY GOSH          20.0000    230.526316   
                          TDS LANDFILL              20.0000    154.732824   
GARBAGE COLLECTIONS       GREAT NORTHERN          8700.0000  19153.333333   
                          HORNSBY BEND            8600.0000   8600.000000   
                          ORGANICS BY GOSH        7500.0000   7500.000000   
                          STEINER LANDFILL         780.0000    780.000000   
                          TDS - MRF               3400.0000  16782.142857   
                          TDS LANDFILL               0.1164  17619.178659   
                          WESTFIELD              16020.0000  16020.000000   
LITTER                    TDS - MRF                360.0000    360.000000   
                          TDS LANDFILL             600.0000   3208.000000   
MIXED LITTER              TDS LANDFILL               0.0000   3196.899696   
ORGANICS                  HORNSBY BEND            6440.0000   8080.000000   
                          ON ROUTE HIGH DUMP      4060.0000  12525.714286   
                          ORGANICS BY GOSH          20.0000  11855.961848   
                          PARK BEND               8480.0000  13385.000000   
                          TDS LANDFILL              20.0000   1273.636364   
RECYCLED METAL            AUSTIN IRON AND METAL    140.0000    451.200000   
                          MRF                      230.0000    230.000000   
                          ORGANICS BY GOSH        5160.0000  12532.000000   
                          TDS LANDFILL             220.0000    220.000000   
RECYCLING - SINGLE STREAM AUSTIN WOOD RECYCLING   2340.0000   4050.000000   
                          BALCONES RECYCLING       140.0000   9464.652072   
                          BALCONES WOODS          8460.0000   8460.000000   
                          TDS - MRF                 40.0000   9809.482345   
                          TDS LANDFILL             420.0000   2580.000000   
SWEEPING                  GREAT NORTHERN                NaN           NaN   
                          ON ROUTE HIGH DUMP            NaN           NaN   
                          ONION CREEK                   NaN           NaN   
                          ORGANICS BY GOSH       14460.0000  15200.000000   
                          PARK CENTER                   NaN           NaN   
                          ST. EDWARDS                   NaN           NaN   
                          TDS LANDFILL             760.0000   9586.815642   
                          WILCAB                        NaN           NaN   
TIRES                     MRF                       40.0000   1664.648485   
                          TDS LANDFILL             380.0000   3512.000000   
YARD TRIMMING             HORNSBY BEND            1220.0000  10981.897810   

                                                      max  
load_type                 dropoff_site                     
BRUSH                     HORNSBY BEND            19660.0  
                          TDS LANDFILL            11860.0  
BULK                      MRF                     15080.0  
                          TDS - MRF               15560.0  
                          TDS LANDFILL            28000.0  
CONTAMINATED ORGANICS     TDS LANDFILL            25380.0  
CONTAMINATED RECYCLING    TDS LAN

In [25]:
# Ekstraksi tanggal dari kolom InvoiceDate
ds['report_date'] = pd.to_datetime(ds['report_date'])
ds['Date'] = ds['report_date'].dt.date
ds

report_date                       load_time load_type  load_weight  \
0      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
1      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
2      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
3      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
4      2021-01-02  2021-01-02 09:24:00.000000 UTC     TIRES       1640.0   
...           ...                             ...       ...          ...   
26152  2021-07-09  2021-07-09 09:48:00.000000 UTC     BRUSH       8240.0   
26153  2021-07-09  2021-07-09 12:02:00.000000 UTC     BRUSH       4160.0   
26154  2021-07-09  2021-07-09 11:55:00.000000 UTC     BRUSH       4340.0   
26155  2021-07-09  2021-07-09 12:00:00.000000 UTC    LITTER       1900.0   
26156  2021-07-09  2021-07-09 12:00:00.000000 UTC    LITTER       3040.0   

       dropoff_site        Date  
0       ONION CREEK  2021-01-01  
1       ONION CREEK  2021-01-01  
2       ONION CREEK  2021-01-01  
3       ONION CREEK  2021-01-01  
4               MRF  2021-01-02  
...             ...         ...  
26152  HORNSBY BEND  2021-07-09  
26153  HORNSBY BEND  2021-07-09  
26154  HORNSBY BEND  2021-07-09  
26155  TDS LANDFILL  2021-07-09  
26156  TDS LANDFILL  2021-07-09  

[26157 rows x 6 columns]

In [26]:
# Group by berdasarkan tanggal dan quantity
sampah_qty = ds[['Date','load_weight']].groupby('Date').sum()
sampah_qty

load_weight
Date                   
2021-01-01          0.0
2021-01-02    2170130.0
2021-01-04    2920260.0
2021-01-05    2388540.0
2021-01-06    2396550.0
...                 ...
2021-07-05    1897290.0
2021-07-06    1921640.0
2021-07-07    1553080.0
2021-07-08    1417650.0
2021-07-09     120340.0

[166 rows x 1 columns]

In [27]:
sampah_qty.describe()

load_weight
count  1.660000e+02
mean   1.849465e+06
std    9.620680e+05
min    0.000000e+00
25%    1.721535e+06
50%    2.133960e+06
75%    2.444950e+06
max    3.745180e+06

In [28]:
print('Rata-Rata =', sampah_qty['load_weight'].mean())
print('Maksimum =', sampah_qty['load_weight'].max())
print('Minimum=', sampah_qty['load_weight'].min())

Rata-Rata = 1849464.852646988
Maksimum = 3745180.0
Minimum= 0.0


Dalam 7 bulan ke belakang, jumlah sampah yang terkumpul setiap harinya mulai dari 0 - 3745 ton dengan rerata 1849 ton.

Menghitung Confidence Interval
\# Asumsikan bahwa data populasi terdistribusi normal
stats.norm.interval(conf_level, loc=average, scale="lebar_distribusi dari average"

In [31]:
std = sampah_qty['load_weight'].std()
N = len(sampah_qty)
stats.norm.interval(0.95, loc=sampah_qty['load_weight'].mean(), scale=std/np.sqrt(N))

(1703112.463669123, 1995817.2416248529)

Diperkirakan jumlah sampah yang akan ditampung setiap harinya 1700 ton hingga 1995 ton.

Rekomendasi : Buat tempat penampungan sampah yang memuat setidaknya 2000 ton sampah.

In [35]:
sampah_type = ds[['load_type','load_weight']].groupby('load_type').sum().sort_values(by='load_weight', ascending=False)
sampah_type

load_weight
load_type                              
GARBAGE COLLECTIONS        1.451478e+08
ORGANICS                   6.793037e+07
RECYCLING - SINGLE STREAM  6.495443e+07
BULK                       1.294074e+07
BRUSH                      9.439680e+06
SWEEPING                   3.462480e+06
YARD TRIMMING              1.504520e+06
MIXED LITTER               1.051780e+06
TIRES                      3.097870e+05
LITTER                     1.286800e+05
RECYCLED METAL             8.567000e+04
CONTAMINATED ORGANICS      2.538000e+04
DEAD ANIMAL                2.465000e+04
CONTAMINATED RECYCLING     5.160000e+03

Manakah jenis tempat penampungan yang sebaiknya dibuat? Jenis garbage collection atau organics?

In [36]:
ds['Bulan'] = ds['report_date'].dt.month
ds

report_date                       load_time load_type  load_weight  \
0      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
1      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
2      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
3      2021-01-01  2021-01-01 12:00:00.000000 UTC  SWEEPING          NaN   
4      2021-01-02  2021-01-02 09:24:00.000000 UTC     TIRES       1640.0   
...           ...                             ...       ...          ...   
26152  2021-07-09  2021-07-09 09:48:00.000000 UTC     BRUSH       8240.0   
26153  2021-07-09  2021-07-09 12:02:00.000000 UTC     BRUSH       4160.0   
26154  2021-07-09  2021-07-09 11:55:00.000000 UTC     BRUSH       4340.0   
26155  2021-07-09  2021-07-09 12:00:00.000000 UTC    LITTER       1900.0   
26156  2021-07-09  2021-07-09 12:00:00.000000 UTC    LITTER       3040.0   

       dropoff_site        Date  Bulan  
0       ONION CREEK  2021-01-01      1  
1       ONION CREEK  2021-01-01      1  
2       ONION CREEK  2021-01-01      1  
3       ONION CREEK  2021-01-01      1  
4               MRF  2021-01-02      1  
...             ...         ...    ...  
26152  HORNSBY BEND  2021-07-09      7  
26153  HORNSBY BEND  2021-07-09      7  
26154  HORNSBY BEND  2021-07-09      7  
26155  TDS LANDFILL  2021-07-09      7  
26156  TDS LANDFILL  2021-07-09      7  

[26157 rows x 7 columns]

In [42]:
#Total jenis sampah 
gar_coll = ds[ds['load_type'] == 'GARBAGE COLLECTIONS'][['Bulan', 'load_weight']].groupby('Bulan').sum()
organics = ds[ds['load_type'] == 'ORGANICS'][['Bulan', 'load_weight']].groupby('Bulan').sum()

In [43]:
print('Rata-rata garbage collection =', gar_coll['load_weight'].mean())
print('Rata-rata organics =', organics['load_weight'].mean())

Rata-rata garbage collection = 20735406.077057146
Rata-rata organics = 9704338.0


Bahwa garbage collection selama 7 bulan terakhir lebih banyak dibandingkan dengan sampah organics.

Melakukan Uji hipotesis Two Sample Independent
--> karena ingin menguji perbedaan signifikan "data populasi" antara garbage collection dengan organics dari data sampel 7 bulan kebelakang.

H0 : Rata-rata garbage collection <= organics

H1 : Rata-rata garbage collection >  organics

In [47]:
t_stat, p_val = stats.ttest_ind(gar_coll['load_weight'], organics['load_weight'])
print('T-Statistic:', t_stat)
print('P-value:', p_val/2) # --> dibagi 2 karena one sided

T-Statistic: 3.046838266938316
P-value: 0.005072071583233548


critical value = 0.05

p-value = 0.005 < critical value

Hasil : H0 ditolak

Dari pengujian secara statsistik diperoleh bahwa jumlah sampah garbage collection dan organics sangat berbeda. Sampah garbage collection akan lebih banyak ditampung dibandingkan sampah irganics. Sehingga akan lebih bagus apabila pembuatan tempat penampungan yang baru yakni jenis garbage collection melihat trend yang cukup besar. 